src/i_oalsound.c

```diff
@@ -610,19 +610,20 @@ static boolean IsPaddedSound(const byte *data, int size)
     return true;
 }
 
-static void FadeInMono8(byte *data, ALsizei size, ALsizei freq)
+static void FadeInOutMono8(byte *data, ALsizei size, ALsizei freq)
 {
     const int fadelen = freq * FADETIME / 1000000;
     int i;
 
-    if (data[0] == 128 || size < fadelen)
+    if ((data[0] == 128 && data[size - 1] == 128) || size < fadelen)
```

Comment:
I'd prefer to distinguish the "needs fade in" from the "neds fade out" case in this check, but then I'd have to go through the loop over `fadelen` twice in the end.

Reply:
I think it's okay to do two loops. It's only looping over 11 or 22 or 44 samples and I'd rather not touch sounds that are already fine at the start or end.